internal/pkg/agent/vault/aesgcm_test.go

```diff
@@ -0,0 +1,211 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package vault
+
+import (
+	"crypto/aes"
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var testKeyTypes []AESKeyType = []AESKeyType{AES128, AES192, AES256}
+
+func TestNewKey(t *testing.T) {
+	for _, kt := range testKeyTypes {
+		b, err := NewKey(kt)
+		if err != nil {
+			t.Error(err)
+		}
+
+		diff := cmp.Diff(int(kt), len(b))
+		if diff != "" {
+			t.Error(diff)
+		}
```

Comment:
Will leave as is. It's a matter of preference, cmp.Diff is used for many things. This way the code is consistent always relying on Diff vs comparing differently and prints out standard diff output. The suggestion is wrong since the diff was not initialized and now you would have to format the error message pointing at the differences manually.